compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/SkiaBasedOwner.skiko.kt

```diff
@@ -237,8 +237,26 @@ internal class SkiaBasedOwner(
         }
     }
 
-    var contentSize = IntSize.Zero
-        private set
+    /**
+     * Provides a way to measure Owner's content in given [constraints]
+     * Draw/pointer and other callbacks won't be called here like in [measureAndLayout] functions
+     */
+    internal fun measureInConstraints(constraints: Constraints): IntSize {
+        val oldConstraints = this.constraints
+        try {
+            // TODO: is it possible to measure without reassigning root constraints?
```

Comment:
I couldn't find a better way to measure in given constraints...

Reply:
Also, as I understand it will break layout caches...

Reply:
But it should be called on parent relayout which should happen a lot

Reply:
I checked, and Swing doesn't call getPreferredSize on window resize in the `initial panel size of LazyColumn with border layout` test. So, we are safe, and we only have an additional measure at startup?

Reply:
It depends on LayoutManager, in case of window resize everything should be fine, since PreferredSize is used only for initial size. But other LayoutManagers may call preferredSize any time :(